src/core/VarReplay.jl

```diff
@@ -70,7 +71,7 @@ mutable struct VarInfo
     vals        ::    Vector{Real}
     rvs         ::    Dict{Union{VarName,Vector{VarName}},Any}
     dists       ::    Vector{Distributions.Distribution}
-    gids        ::    Vector{Int}
+    gids        ::    Vector{Set{Selector}}
```

Comment:
Why is it a set? IIUC, every variable can only belong to one sampler, hence one gid is enough, no?

Reply:
This relaxed the assumption the one-to-one mapping between variables and samplers. In general, variables can be mapped to multiple component samplers in a Gibbs sampler. For example, a variable can be updated by: `Gibbs(PG(:z), HMC(:z))`. The result will still be correct and can lead to performance gains for some elaborate models.

Reply:
I see, thanks for the explanation.